static/src/javascripts/projects/commercial/initialConsentState.ts

```diff
@@ -0,0 +1,39 @@
+import { onConsentChange } from '@guardian/consent-management-platform';
+import type { ConsentState } from '@guardian/consent-management-platform/dist/types';
+
+/**
+ * Return a promise containing the first consent state provided by the user
+ * as soon as it becomes available. This will only resolve once whereas
+ * callbacks passed to onConsentChange are executed each time consent state changes.
+ * NOTE: depending on where this function is eventually used, it might be more appropriate
+ * for it to be defined in the consent-management-platform
+ * @returns Promise
+ */
+export const getInitialConsentState = (): Promise<ConsentState> => {
```

Comment:
Maybe something like this?

```suggestion
/**
 * A promise that returns *only* the first consent state from the user.
 * 
 * Because we cannot “unload” third-party scripts, listening to further
 * changes in consent via `onConsentState` can yield unexpected
 * results. Avoid using this method if your code can react to subsequent calls.
 */
export const getInitialConsentState = (): Promise<ConsentState> => {
```

I would be good to check what the percentage of page view is where a user goes from a little to more consent. I expect the opposite to happen more often.

Reply:
I'm a bit torn about including info about when and where the function is used (for now, loading third-party scripts) in the function's doc string rather than sticking to what it does (returning only the initial consent state).
The former is bound to get out of date relatively quickly as we add and subtract call sites whereas the latter is easier to keep current.

Reply:
That’s a fair point. I think we should warn that this is only to be used in cases where the code cannot react to subsequent calls (as in third-party scripts loading).

It should also mention in TCFv2 we’ll skip the `cmpuishown` state. It will only resolve once we have a user action.

Reply:
How'd that?

Reply:
```ts
if (state.tcfv2.eventStatus !== 'cmpuishown')
  resolveInitialState(state);
else return;
```